src/compiler/crystal/semantic/call.cr

```diff
@@ -467,7 +467,7 @@ class Crystal::Call
 
   def named_tuple_indexer_helper(args, arg_types, owner, instance_type, nilable)
     arg = args.first
-    if arg.is_a?(SymbolLiteral)
+    if arg.is_a?(SymbolLiteral) || arg.is_a?(StringLiteral)
```

Comment:
Not sure this will filter `arg` to the correct type.

In any case, really nice that you touched compiler code! 🎉